pkg/ctl/install/cmd.go

```diff
@@ -0,0 +1,100 @@
+package install
+
+import (
+	"context"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+)
+
+// Command will create the `install` commands
+func Command(flagGrouping *cmdutils.FlagGrouping) *cobra.Command {
+	verbCmd := cmdutils.NewVerbCmd("install", "Install components in a cluster", "")
+
+	cmdutils.AddResourceCmd(flagGrouping, verbCmd, installFluxCmd)
+
+	return verbCmd
+}
+
+type installFluxOpts struct {
+	gitURL              string
+	gitBranch           string
+	gitPaths            []string
+	gitLabel            string
+	gitUser             string
+	gitEmail            string
+	gitFluxPath         string
+	namespace           string
+	tillerNamespace     string
+	timeout             time.Duration
+	amend               bool
+	noHelmOp            bool
+	noTiller            bool
+	tillerHost          string
+	helmOpTLSCertFile   string
+	helmOpTLSKeyFile    string
+	helmOpTLSCACertFile string
+}
+
+func installFluxCmd(cmd *cmdutils.Cmd) {
+	cmd.ClusterConfig = api.NewClusterConfig()
+	cmd.SetDescription(
+		"flux",
+		"Bootstrap Flux, installing it in the cluster and initializing its manifests in the specified Git repository",
+		"",
+	)
+	var opts installFluxOpts
+	cmd.SetRunFuncWithNameArg(func() error {
+		installer, err := newFluxInstaller(context.Background(), cmd, &opts)
+		if err != nil {
+			return err
+		}
+		return installer.run(context.Background())
+	})
+	cmd.FlagSetGroup.InFlagSet("Flux installation", func(fs *pflag.FlagSet) {
+		fs.StringVar(&opts.gitURL, "git-url", "",
+			"URL of the Git repository to be used by Flux, e.g. git@github.com:<your username>/flux-get-started")
+		fs.StringVar(&opts.gitBranch, "git-branch", "master",
+			"Git branch to be used by Flux")
+		fs.StringSliceVar(&opts.gitPaths, "git-paths", []string{},
+			"Relative paths within the Git repo for Flux to locate Kubernetes manifests")
+		fs.StringVar(&opts.gitLabel, "git-label", "flux",
+			"Git label to keep track of Flux's sync progress; overrides both --git-sync-tag and --git-notes-ref")
+		fs.StringVar(&opts.gitUser, "git-user", "Flux",
+			"Username to use as Git committer")
+		fs.StringVar(&opts.gitEmail, "git-email", "",
+			"Email to use as Git committer")
+		fs.StringVar(&opts.gitFluxPath, "git-flux-subdir", "flux/",
+			"Directory within the Git repository where to commit the Flux manifests")
+		fs.StringVar(&opts.namespace, "namespace", "flux",
+			"Cluster namespace where to install Flux and the Helm Operator")
+		fs.StringVar(&opts.tillerNamespace, "tiller-namespace", "kube-system",
```

Comment:
Is there a requirement for this? I'd put it into same namespace where flux is, avoid having a flag, and restrict it to use just that namespace.

Reply:
There is no requirement, but tiller is usually installed in `kube-system`.

If we put it in the same namespace then we don't need to make tiller optional, which would probably make things simpler though.

@kalbir thoughts?

Reply:
I can see reasons for doing it either way. I like the idea of putting everything we install into our "own" namespaces rather than polluting `kube-system` but I have no strong opinion on what users would expect or prefer.

Reply:
My idea was to make it more flexible and less opinionated by reusing an existing Tiller installation if the user wanted to. But then again, this is probably about novice users. So, I am on the fence about this one.

Reply:
Ok, I think we should reverse that. There's is no point, many options of installing these components are available to the user, we only provide a convenient shortcut for most part. We must enable higher level use-cases, and ungrnore low-level knobs whenever possible until multiple users request a particular kind of a knob.

Reply:
Alright. I will always install tiller.